src/composables/trade/useSor.ts

```diff
@@ -245,17 +245,28 @@ export default function useSor({
           tokenOutAddress.toLowerCase()
         );
 
-        const tokenInAmount = BigNumber.from(deltas[tokenInPosition]).abs();
-
-        const tokenOutAmount = BigNumber.from(deltas[tokenOutPosition]).abs();
+        let tokenInAmount = BigNumber.from(deltas[tokenInPosition]).abs();
+        let tokenOutAmount = BigNumber.from(deltas[tokenOutPosition]).abs();
 
         if (swapType === SwapType.SwapExactOut) {
+          tokenInAmount = await mutateAmount({
```

Comment:
I don't think these two if statements are needed. Otherwise a exact in stETH trade amount won't be mutated.

Reply:
This part is about mutating the output token only. It's called every block to make sure the display number for the amount you are not specifying is kept up to date. I missed performing the mutations on the values for the priceImpact further down in the `handleAmountChange` function, which catches the case you are talking about.